### ISSUE TITLE
[BUG] [Java] Java generator initialize non-required collections with empty collection 

### ISSUE THREAD

**Issue by Giletka** [OpenAPITools/openapi-generator#14875]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

After upgrading openapi generator 5.1.1 -> 6.3.0 our integration tests was broken. Reason that API expected null as a init value of the non-required collection property in the pojo classes, but new version initialize it with empty collection (like required ones). 

Expected:
```java
  public static final String JSON_PROPERTY_EMAILS = "emails";
  private List<ScimUserEmails> emails = new ArrayList<>();

  public static final String JSON_PROPERTY_ADDRESSES = "addresses";
  private List<ScimUserAddresses> addresses = null;
```

Actual:
```java
  public static final String JSON_PROPERTY_EMAILS = "emails";
  private List<ScimUserEmailsInner> emails = new ArrayList<>();

  public static final String JSON_PROPERTY_ADDRESSES = "addresses";
  private List<ScimUserAddressesInner> addresses = new ArrayList<>();
```

Also, generator don't respect `@Nullable` annotation, but works as expected if you add `nullable: true` in spec to the property

##### openapi-generator version

6.2.1, 6.3.0, 6.4.0

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.2
info:
  title: 'Shared SCIM object definitions'
  description: 'SCIM object definitions'
  version: 1.0.1
externalDocs:
  description: Find out more about CIS
  url: https://where.am.i
servers:
  - url: https://where.am.i
paths:
  /test:
    post:
      summary: Test a user with email and address
      operationId: test
      responses:
        '200':
          description: 'Some description'
          content:
            application/scim+json:
              schema:
                $ref: '#/components/schemas/ScimUser'
components:
  schemas:
    ScimUser:
      type: object
      description: 'Here is some description'
      required:
        - userName
        - emails
      properties:
        userName:
          type: string
        emails:
          type: array
          items:
            type: object
            properties:
              key:
                type: string
              type:
                type: string
              value:
                type: string
              display:
                type: string
              primary:
                type: boolean
        addresses:
          type: array
          items:
            type: object
            properties:
              key:
                type: string
              type:
                type: string
              display:
                type: string
              primary:
                type: boolean
              country:
                type: string
              formatted:
                type: string
              locality:
                type: string
              postalCode:
                type: string
              region:
                type: string
              streetAddress:
                type: string
```

##### Generation Details

Used Java 11 with gradle
```groovy
task generateScimModels(type: GenerateTask) {
  generatorName = "java"
  inputSpec = "${inputSpecBasePath}/spec-scim.yaml"
  outputDir = "${buildDir}/generated/sources/scim"
  generateModelTests = false
  generateModelDocumentation = false
  generateApiTests = false
  generateApiDocumentation = false
  validateSpec = false
  apiFilesConstrainedTo = []
  supportingFilesConstrainedTo = []
  modelFilesConstrainedTo = [
      "ScimUser"
  ]
  configOptions = [
      dateLibrary                    : "java8",
      library                             : "resttemplate",
      modelPackage               : "ecg.identity.scim.model",
      useBeanValidation        : "false",
      performBeanValidation: "false",
      supportingFiles             : "false",
      serializableModel          : "true"
  ]
}
```

##### Steps to reproduce

1. Create pojo in spec with required and non-required collections as properties
2. Generate with "java" generator

##### Related issues/PRs
This one provides toggle to set null as default for all collections (required and non-required). But no solution to keep null only for non-required as in previous versions.
#1861 

##### Suggest a fix

For me I just made this workaround in openapi/Java/pojo.mustache.
Changed this
```
  {{^vendorExtensions.x-is-jackson-optional-nullable}}
  {{#isContainer}}
  private {{{datatypeWithEnum}}} {{name}}{{#defaultValue}} = {{{.}}}{{/defaultValue}};
  {{/isContainer}}
```
  
To this
```  
  {{^vendorExtensions.x-is-jackson-optional-nullable}}
  {{#isContainer}}
  private {{{datatypeWithEnum}}} {{name}} = {{#required}}{{{defaultValue}}}{{/required}}{{^required}}null{{/required}};
  {{/isContainer}}
```

**Comment by wing328** [comment]
have you tried setting the `java` generator option `containerDefaultToNull` to `true` ?

**Comment by Giletka** [comment]
> have you tried setting the `java` generator option `containerDefaultToNull` to `true` to see if that meets your requirement?

Yes, this option makes all collections initialize as null, but default behavior expected empty collection in requred ones
I wrote about it here (sorry if it wasn't clear):

> This one provides toggle to set null as default for all collections (required and non-required). But no solution to keep null only for non-required as in previous versions.
https://github.com/OpenAPITools/openapi-generator/issues/1861

**Comment by Giletka** [comment]
Also, I want to add that "spring" generator, for example, works as expected: required ones generated with empty collection, non-required with null.

**Comment by eivbjorkelund** [comment]
Having the same issues when we upgrade to 6.3.0 and 6.4.0. It works fine in 6.2.1.

**Comment by wing328** [comment]
Hi all, can you please test https://github.com/OpenAPITools/openapi-generator/pull/14961 when you've time?

I did a test with the spec provided in this issue and the result looks good:
```
/**
 * Here is some description
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", date = "2023-03-15T15:56:12.256532+08:00[Asia/Hong_Kong]")
public class ScimUser {
  public static final String SERIALIZED_NAME_USER_NAME = "userName";
  @SerializedName(SERIALIZED_NAME_USER_NAME)
  private String userName;

  public static final String SERIALIZED_NAME_EMAILS = "emails";
  @SerializedName(SERIALIZED_NAME_EMAILS)
  private List<ScimUserEmailsInner> emails = new ArrayList<>();

  public static final String SERIALIZED_NAME_ADDRESSES = "addresses";
  @SerializedName(SERIALIZED_NAME_ADDRESSES)
  private List<ScimUserAddressesInner> addresses;
```

**Comment by wing328** [comment]
@eivbjorkelund thanks for confirming the fix 🙏

**Comment by wing328** [comment]
the fix has been merged and will be included in the upcoming 6.5.0 release

**Comment by wing328** [comment]
@fastluca please open a new issue with a spec to reproduce the issue.
